_install-and-configure/upgrade-opensearch/appendix/rolling-upgrade-lab.md

```diff
@@ -0,0 +1,887 @@
+---
+layout: default
+title: Rolling Upgrade - Lab
+parent: Upgrades Appendix
+grand_parent: Upgrading OpenSearch
+nav_order: 50
+redirect_from:
+  - /upgrade-opensearch/appendix/rolling-upgrade-lab/
+---
+
+<!--
+Testing out tabs for code blocks to identify example outputs and file names.
+To use, invoke class="codeblock-label"
+-->
+
+<style>
+.codeblock-label {
+    display: inline-block;
+    border-top-left-radius: 0.5rem;
+    border-top-right-radius: 0.5rem;
+    font-family: Menlo,Monaco,Consolas,Liberation Mono,Courier New,monospace;
+    font-size: .75rem;
+    --bg-opacity: 1;
+    background-color: #e1e7ef;
+    background-color: rgba(224.70600000000002,231.07080000000002,239.394,var(--bg-opacity));
+    padding: 0.25rem 0.75rem;
+    border-top-width: 1px;
+    border-left-width: 1px;
+    border-right-width: 1px;
+    --border-opacity: 1;
+    border-color: #ccd6e0;
+    border-color: rgba(204,213.85999999999999,224.39999999999998,var(--border-opacity));
+    margin-bottom: 0;
+}
+</style>
+
+# Rolling Upgrade - Lab
+
+You can follow these steps on your own compatible host to recreate the same cluster state the OpenSearch Project used for testing [rolling upgrades]({{site.url}}{{site.baseurl}}/install-and-configure/upgrade-opensearch/rolling-upgrade/). This exercise is useful if you want to test the upgrade process in a development environment.
+
+The steps used in this lab were validated on an arbitrarily-chosen [Amazon Elastic Compute Cloud (Amazon EC2)](https://aws.amazon.com/ec2/) `t2.large` instance using [Amazon Linux 2](https://aws.amazon.com/amazon-linux-2/) kernel version `Linux 5.10.162-141.675.amzn2.x86_64` and [Docker](https://www.docker.com/) version `20.10.17, build 100c701`. The instance was provisioned with an attached 20 GiB gp2 [Amazon Elastic Block Store (EBS)](https://aws.amazon.com/ebs/) root volume. These specifications are included for informational purposes and do not represent hardware requirements for OpenSearch or OpenSearch Dashboards.
```

Comment:
Removing "arbitrarily-chosen" here. 

```suggestion
The steps used in this lab were validated on an arbitrarily-chosen [Amazon Elastic Compute Cloud (Amazon EC2)](https://aws.amazon.com/ec2/) `t2.large` instance using [Amazon Linux 2](https://aws.amazon.com/amazon-linux-2/) kernel version `Linux 5.10.162-141.675.amzn2.x86_64` and [Docker](https://www.docker.com/) version `20.10.17, build 100c701`. The instance was provisioned with an attached 20 GiB gp2 [Amazon Elastic Block Store (EBS)](https://aws.amazon.com/ebs/) root volume. These specifications are included for informational purposes and do not represent hardware requirements for OpenSearch or OpenSearch Dashboards.
```

Reply:
Follow up question, have the instructions below only been tested with the instance described here? Are there any other configurations where these tests have been tested?

Reply:
Can you define "other configurations?"

Reply:
I'm including the phrase "arbitrarily-chosen" because I'm anticipating questions from discerning readers. The language is there to reinforce that a reader should not focus on the underlying host details because they are portable to any compatible environment.

Reply:
As in besides `t2-large` or OS besides Amazon Linux 2? 

Also, I would avoid use of the word "arbitrary", because doesn't sound friendly or confident. While it might be accurate to how chose the EC2 options for your lab, it leads me to think that the choices I make at the beginning of the lab doesn't matter, even though OpenSearch itself does require a fairly sizable amount of CPU-power to run.

Reply:
@natebower: Any thoughts on this one?

Reply:
Do we mean "randomly"?

Reply:
No, not random - arbitrary, to prevent the question of "why did you do it on platform xyz?." Needs to be taken into context with the rest of the section.

Removing the hyphen.